galaxy_ng/app/management/commands/set-repo-keyring.py

```diff
@@ -67,9 +68,15 @@ def handle(self, *args, **options):
                     self.echo("Process canceled.")
                     return
 
+        proc = subprocess.run([
+            "gpg", "--keyring", keyring_path, "--export", "-a"
```

Comment:
I'd suggest setting `--homedir` to a tmpdir to avoid any sideeffects from usersettings.

Also there might be a solution using `import gnupg; gnupg.GPG(gnupghome=tmpdir)`.